src/Microsoft.Azure.WebJobs.Logging.ApplicationInsights/ApplicationInsightsLogger.cs

```diff
@@ -459,31 +463,99 @@ private void StartTelemetryIfFunctionInvocation(IDictionary<string, object> stat
                     {
-                        Name = functionName
-                    };
+                        RequestTelemetry request = new RequestTelemetry
+                        {
+                            Name = functionName
+                        };
 
-                    // We'll need to store this operation context so we can stop it when the function completes
-                    IOperationHolder<RequestTelemetry> operation = _telemetryClient.StartOperation(request);
+                        operation = _telemetryClient.StartOperation(request);
 
+                        // if there is more than one link (batch dispatch in EventHub trigger)
+                        // we'll populate link information on the request telemetry, but don't touch parent for this request
+                        PopulateLinks(activities, request);
+                    }
+
+                    // We'll need to store this operation context so we can stop it when the function completes
                     stateValues[OperationContext] = operation;
                 }
             }
         }
 
+        private void PopulateLinks(Activity[] activities, RequestTelemetry request)
+        {
+            if (activities == null)
+            {
+                return;
+            }
+
+            if (activities.Any(l => l.Recorded))
+            {
+                request.ProactiveSamplingDecision = SamplingDecision.SampledIn;
+            }
+
+            var linksJson = new StringBuilder();
+            linksJson.Append('[');
+            foreach (var link in activities)
+            {
+                var linkTraceId = link.TraceId.ToHexString();
+
+                // avoiding json serializers for now for the sake of perf.
+                // serialization is trivial and looks like `_MS.links` property with json blob
+                // [{"operation_Id":"5eca8b153632494ba00f619d6877b134","id":"|5eca8b153632494ba00f619d6877b134.d4c1279b6e7b7c47."},
+                //  {"operation_Id":"ff28988d0776b44f9ca93352da126047","id":"|ff28988d0776b44f9ca93352da126047.bf4fa4855d161141."}]
+                linksJson
+                    .Append('{')
+                    .Append("\"operation_Id\":")
+                    .Append('\"')
+                    .Append(linkTraceId)
+                    .Append('\"')
+                    .Append(',');
+                linksJson
+                    .Append("\"id\":")
+                    .Append('\"')
+                    .Append('|')
+                    .Append(linkTraceId)
+                    .Append('.')
+                    .Append(link.ParentSpanId.ToHexString())
+                    .Append('.')
+                    .Append('\"');
+
+                // we explicitly ignore sampling flag, tracestate and attributes at this point.
+                linksJson.Append("},");
+            }
+
+            linksJson.Append("]");
+
+            request.Properties["_MS.links"] = linksJson.ToString();
```

Comment:
is this a special property?

Reply:
Yeah, UX will use this property to discover and show many-to-one relationship (many messages are merging into one processing). 

'_MS' is prefix that indicates internal stuff and 'links' is a common name in tracing world to represent such kind of relationships in batching scenarios